### DISCUSSION TITLE
支持 checksums 以方便校验文件上传是否完整

### DISCUSSION THREAD

**Discussion by muyinliu** [messense/aliyundrive-webdav#448] [Q&A] [answered]
1. 阿里云盘 API https://api.aliyundrive.com/adrive/v3/file/list 返回的数据 `.items[n].content_hash` 实际为文件内容的 SHA-1
2. 支持 checksums 后可以方便校验上传的文件是否完整（参考 [Nextcloud checksums](https://docs.nextcloud.com/server/14/developer_manual/client_apis/WebDAV/basic.html#requesting-properties) `{http://owncloud.org/ns}checksums` 部分 `<oc:checksums />`、[webdav: support MD5 and SHA1 hashes with Owncloud and Nextcloud - fixes - rclone](https://github.com/rclone/rclone/commit/8a774a3dd47cfe499e3ceaacaca3ac442e5a99c1)）

**Comment by messense** [comment]
个人没有上传需求，欢迎自行 PR 实现，或者[付费技术支持](https://github.com/messense/aliyundrive-webdav/discussions/442)。

**Comment by messense** [comment, marked as answer]
Implemented in https://github.com/messense/aliyundrive-webdav/pull/524
